_posts/en/newsletters/2020-04-29-newsletter.md

```diff
@@ -0,0 +1,288 @@
+      (for example, sending the pinned preimage settlement transaction
+      to known miners but sending a different non-settlement transaction
+      with at least one of the same inputs to non-miner relay nodes).
+
+    - **Beg or pay for preimages:** The relay network could provide
+      [information about conflicts][harding reject] to transaction
+      submitters such as Bob so they wouldn't need to continuously
+      monitor relay themselves.  This still suffers from the problem of
+      bad actors such as Mallory using [targeted relay][corallo targeted
+      relay] to send different transactions to miners and non-miners.
+      Additionally Bob might be able to [pay][harding pay] miners or
+      other third party nodes for the preimage he needs, although this
+      requires some people run additional software and [might not be as
+      easy][zmn ptlcs] to do after the deployment of some proposed
+      upgrades to the LN protocol.
+
+    - **Settlement transaction anchor outputs:** Onchain settlement
+      transactions could be redesigned to spend their value to [anchor
+      outputs][topic anchor outputs] that could be [CPFP fee
+      bumped][topic cpfp] using [CPFP carve-out][topic cpfp carve out].
+      This would require those transactions to be larger (increasing
+      onchain fees) and presigned (reducing flexibility).  This would
+      only directly affect channels which are unilaterally closed while payments
+      are pending, which is already a situation that that can
+      significantly increase onchain costs and so is something users try
+      to avoid.  However, raising the cost of onchain enforcement also
+      raises the minimum value of payments that can be sent trustlessly
+      through LN with economical enforcement.  Depsite these challenges,
+      as of this writing, this appears to be the most preferred
+      solution.
+
+    Corallo labeled this a severe issue but noted its similar
+    consequences to another known issue related to fee management in
+    onchain LN transactions.  The existing issue (described in
+    [Newsletter #78][news78 anchor outputs]) is that commitment
+    transactions have their feerate set at the time the transaction is
+    signed, which might be days or weeks before they're broadcast to
+    Bitcoin relay nodes.  If the minimum feerate necessary to get a
+    transaction included in the next few blocks has increased
+    significantly since the transaction was last signed, then the
+    commitment transaction might not confirm until after Alice is able
+    to reclaim her funds from Bob, again creating the opportunity for
+    Bob to end up both paying Mallory and giving a refund to Alice.
+    (This existing issue is what developers were working on fixing when
+    the new issue was discovered.  Its ultimate solution also depends on
+    full nodes being able to perform [package relay][Bitcoin Core
+    #14895].)
```

Comment:
Possibly worth mentioning that discussions regarding this issue have been ongoing for some time now (iirc, since 1 or 2 years at a general level, and somewhat acutely the past 6 months).

Reply:
I'd say acutely for at least 2 years :). In any case, it may be worth downplaying the package relay requirement a bit here - not that it isn't eventually required but that, in today's mempool, we don't see any issues here, and if lightning nodes hardcoded 2 sat/vbyte trends dont seem to indicate its a critical issue immediately.

Reply:
> I'd say acutely for at least 2 years :)

Hehe :)